prusti-common/src/vir/ast/expr.rs

```diff
@@ -867,6 +870,7 @@ impl Expr {
                 match constant {
                     Const::Bool(..) => &Type::Bool,
                     Const::Int(..) | Const::BigInt(..) => &Type::Int,
+                    Const::Dummy => &Type::Int,
```

Comment:
Why is this an `Int`? If it just a temporary workaround, would be good to add a comment explaining that.

Reply:
I chose `Int` as a "simple" type. Maybe it could be a `TypedRef` to some unique, abstract instance …

The alternative for passing a dummy value that is always `0` (for example) is to process function signatures such that we completely skip encoding the argument. That would be more work, more likely to result in errors, and maybe it would cause more conflicts with closures, where the value must be passed.

To be clear:

```rust
fn foo<F: Fn...>(x: F) { ... }
```

We can always just pass a dummy value for `x`, because the function is actually identified by the type, and so with monomorphisation and spec. entailments, we have everything we need both at call site and declaration site.

Reply:
I would either pick `TypedRef("FnPtr")` that is always set to Viper `null` or `Domain("FnPtr")` that is set to `fn_ptr()`.